src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1431,6 +1431,10 @@ private BoundExpression BindAssignment(AssignmentExpressionSyntax node, BindingD
                 op2 = BindToNaturalType(op2, diagnostics);
                 op1 = InferTypeForDiscardAssignment((BoundDiscardExpression)op1, op2, diagnostics);
             }
+            else
+            {
+                op1 = BindToTypeForErrorRecovery(op1);
```

Comment:
>op1 = BindToTypeForErrorRecovery(op1); [](http://example.com/codeflow?start=16&length=38)

This doesn't look like a code path that is taken only in an error situation, therefore, calling ```BindToTypeForErrorRecovery``` unconditionally makes me uncomfortable.